man/kernel-install.xml

```diff
@@ -163,36 +175,55 @@
     <para>If <option>--verbose</option> is used, <varname>$KERNEL_INSTALL_VERBOSE=1</varname> will be set for
     the plugins. They may output additional logs in this case.</para>
 
-    <para>If <varname>MACHINE_ID=</varname> is set and not empty, it will be used as <replaceable>MACHINE-ID</replaceable>,
-    overriding any automatic detection attempts.  The value must be a valid machine ID (32 hexadecimal characters).</para>
-
-    <para><varname>KERNEL_INSTALL_MACHINE_ID=</varname> is set for the plugins to the desired <replaceable>MACHINE-ID</replaceable> –
-    either 32 hexadecimal characters or the special value <literal>Default</literal>.</para>
-    <para><varname>KERNEL_INSTALL_BOOT_ROOT=</varname> is set for the plugins to the root directory (mount point, usually) of the hierarchy
-    where boot-loader entries, kernel images, and associated resources should be placed. Can be overridden by setting <varname>BOOT_ROOT=</varname>.</para>
-
-    <para><varname>KERNEL_INSTALL_LAYOUT=bls|other|...</varname> is set for the plugins to specify the installation layout.
-    Defaults to <option>bls</option> if <filename>$BOOT/<replaceable>MACHINE-ID</replaceable></filename> exists, or <option>other</option> otherwise.
-    Additional layout names may be defined by convention. If a plugin uses a special layout,
-    it's encouraged to declare its own layout name and configure <varname>layout=</varname> in <filename>install.conf</filename> upon initial installation.</para>
-
-    <para><varname>KERNEL_INSTALL_INITRD_GENERATOR=...</varname> is set for plugins to select the initrd generator.
-    This should be configured as <varname>initrd_generator=</varname> in <filename>install.conf</filename>.
-    </para>
-
-    <para><varname>KERNEL_INSTALL_STAGING_AREA=...</varname> is set for plugins to a path to a directory.
-    Plugins may drop files in that directory, and they will be installed as part of the loader entry, based
-    on the file name and extension.</para>
+    <para>If <varname>$MACHINE_ID</varname> is set and not empty when <command>kernel-install</command> is
+    invoked, it will be used as <replaceable>MACHINE-ID</replaceable>, overriding any automatic detection
+    attempts.  The value must be a valid machine ID (32 hexadecimal characters).</para>
+
+    <para><varname>$KERNEL_INSTALL_MACHINE_ID</varname> is set for the plugins to the desired
+    <replaceable>MACHINE-ID</replaceable> to use. It's always a 128bit ID, and typically the ID from
+    <filename>/etc/machine-id</filename> or the one passed in via <varname>$MACHINE_ID</varname>.</para>
```

Comment:
I'd add a note here about MACHINE_ID being some random UUID if no machine ID is provided and /etc/machine-id is empty